packages/next-auth/src/providers/hubspot.ts

```diff
@@ -0,0 +1,104 @@
+import type { OAuthConfig, OAuthUserConfig } from "."
+
+interface HubSpotProfile extends Record<string, any> {
+
+  // TODO: figure out additional fields, for now using 
+  // https://legacydocs.hubspot.com/docs/methods/oauth2/get-access-token-information
+
+  user: string,
+  user_id: string,
+
+  hub_domain: string,
+  hub_id: string,
+}
+
+
+const HubSpotConfig = {
+  authorizationUrl: "https://app.hubspot.com/oauth/authorize",
+  tokenUrl: "https://api.hubapi.com/oauth/v1/token",
+  profileUrl: "https://api.hubapi.com/oauth/v1/access-tokens"
+}
+
+export default function HubSpot<P extends HubSpotProfile>(
+  options: OAuthUserConfig<P>
+): OAuthConfig<P> {
+
+  return {
+    id: "hubspot",
+    name: "HubSpot",
+    type: "oauth",
+
+    ...HubSpotConfig,
+
+    authorization: {
+      url: HubSpotConfig.authorizationUrl,
+      params: {
+        scope: "oauth",
+        client_id: options.clientId,
+      },
+
+    },
+    token: {
+      url: HubSpotConfig.tokenUrl,
+      async request(context) {
```

Comment:
@Miciurash, sorry for not suggesting this earlier, but it seems like we can use:
```

    client: {
      token_endpoint_auth_method: "client_secret_post",
    },
    token: HubSpotConfig.tokenUrl,
```

And `node-openid-client` will take care of sending the request. I've also just learned about this in [the other PR](https://github.com/nextauthjs/next-auth/pull/5236) 😅
I've tested it and it works for me locally.

Reply:
Great suggestion @ThangHuuVu! I learned something new today, updated it and it works like a charm!